### PR TITLE
Largest Mars Picture

### PR DESCRIPTION
**Pull request by DanielChanger**

### CODE REVIEW

File: src/main/java/com/changer/marspictures/MarsPicturesController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+@RestController
+@RequestMapping("/mars/pictures")
+@RequiredArgsConstructor
+public class MarsPicturesController {
+
+    private final MarsPicturesService marsPicturesService;
+
+    @GetMapping(value = "/largest", produces = MediaType.IMAGE_JPEG_VALUE)
+    public ResponseEntity<byte[]> findLargestPicture(@RequestParam("sol") int sol,
```
Comment (StanislavZabramnyi) [inline src/main/java/com/changer/marspictures/MarsPicturesController.java:21]: It is not mandatory to specify the name of parameter in value variable if it has the same name as method argument

File: src/main/java/com/changer/marspictures/MarsPicturesController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                                                     @RequestParam(value = "camera", required = false) String camera) {
+        Optional<byte[]> largestPicture = marsPicturesService.findLargestPicture(sol, camera);
+        if (largestPicture.isEmpty()) {
+            return ResponseEntity.notFound().build();
```
Comment (StanislavZabramnyi) [inline src/main/java/com/changer/marspictures/MarsPicturesController.java:25]: See this item from the task description -> "If no pictures found by provided params, it should respond with the error message "No pictures found""

File: src/main/java/com/changer/marspictures/MarsPicturesService.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class MarsPicturesService {
+
+    private final NasaApiInfo nasaApiInfo;
+    private final RestTemplate restTemplate;
+
+    @Cacheable("pictures")
+    public Optional<byte[]> findLargestPicture(int sol, String camera) {
```
Comment (StanislavZabramnyi) [inline src/main/java/com/changer/marspictures/MarsPicturesService.java:25]: you don`t need to load the whole picture in your app memory, you can just get the Pictures URI from the nasaAPI and use redirect to this URI in your controller. smth like this 
```
 ResponseEntity.status(MOVED_PERMANENTLY)
                .header(LOCATION, largestPicture.getPhotoUrl())
                .build();
```

File: src/main/java/com/changer/marspictures/MarsPicturesController.java (hunk 1 above)
Comment (DanielChanger) [reply src/main/java/com/changer/marspictures/MarsPicturesController.java:21]: Yeah, thanks!

File: src/main/java/com/changer/marspictures/MarsPicturesController.java (hunk 2 above)
Comment (DanielChanger) [reply src/main/java/com/changer/marspictures/MarsPicturesController.java:25]: One of the stuff I forgot

Comment (StanislavZabramnyi) [reply src/main/java/com/changer/marspictures/MarsPicturesController.java:25]: By the way, have you seen the time? It's 2:25 a.m. Please go to bed and rest. Nasa pictures can wait)

**Pull request merged**